server/src/main/java/org/elasticsearch/search/fetch/subphase/FieldFetcher.java

```diff
@@ -38,6 +39,11 @@
  */
 public class FieldFetcher {
 
+    /**
+     * Default maximum number of states in the automaton that looks up unmapped fields.
+     */
+    private static final int AUTOMATON_MAX_DETERMINIZED_STATES = 100000;
```

Comment:
For context, I'm wondering if there's reasoning behind choosing 100,000?

Reply:
Not in particular. It was one of the numbers brought up. Should provide ample headroom, fortunetely the automanon grabs memory lazily so this should never use up much space in normal cases.